Duo/duo/connector.py

```diff
@@ -135,6 +135,21 @@ def fetch_batches(self):
         for events in self.get_events_iterator():
             batch_start_time = time.time()
 
+            if len(events) > 0:
+                if self._log_type == LogType.TELEPHONY:
+                    # Telephony logs have their datetime represented as "2023-03-21T22:34:49.466370+00:00"
+                    most_recent_event = max(events, key=lambda item: item["ts"])
+                    most_recent_timestamp = datetime.datetime.fromisoformat(most_recent_event["ts"]).timestamp()
+                else:
+                    most_recent_event = max(events, key=lambda item: item.get("timestamp"))
```

Comment:
If one of the events doesn't contain the field `timestamp`, the max function will generate an exception. I suggest filtering the list of events before.
```suggestion
                    most_recent_event = max(filter(lambda item: item.get("timestamp"), events))
```